test/integration/generated/mediaTypes/src/models/index.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ * Changes may cause incorrect behavior and will be lost if the code is regenerated.
+ */
+
+import * as coreHttp from "@azure/core-http";
+
+/**
+ * Uri or local path to source data.
+ */
+export interface SourcePath {
+  /**
+   * File source path.
+   */
+  source?: string;
+}
+
+/**
+ * Defines values for ContentType.
+ */
+export type ContentType =
+  | "application/pdf"
+  | "image/jpeg"
+  | "image/png"
+  | "image/tiff";
+
+/**
+ * Optional parameters.
+ */
+export interface MediaTypesClientAnalyzeBody$binaryOptionalParams
```

Comment:
Just curious, why do we add '$' before binary? Is that part of a convention?

Reply:
I wanted to add some special character to protect against the off-chance scenario where there's also a `MediaTypesClientAnalyzeBodybinary` operation that would cause a name conflict. I chose `$` since I figured no methods would contain that character, but could have gone with underscore instead.

Reply:
Makes sense thanks for clarifying